gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -850,7 +850,13 @@ public void apply(Project project)
         configurePegasusSchemaSnapshotGeneration(project, sourceSet, false);
       }
 
-      configurePegasusSchemaSnapshotGeneration(project, sourceSet, true);
+      // Do compatibility check if there are extension schemas.
+      if (!SharedFileUtils.getSuffixedFiles(project,  project.file(getExtensionSchemaPath(project, sourceSet)),
```

Comment:
Looking at other methods like `configureRestClientConfiguration`, this sort of conditional check is done inside the method. I'd suggest making it consistent.